src/mscorlib/CreateRuntimeRootILLinkDescriptorFile.targets

```diff
@@ -0,0 +1,28 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <CompileDependsOn>_CreateILLinkRuntimeRootDescriptorFile;$(CompileDependsOn)</CompileDependsOn>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <_ILLinkRuntimeRootDescriptorFileName Condition=" '$(_ILLinkRuntimeRootDescriptorFileName)' == '' ">System.Private.CoreLib.xml</_ILLinkRuntimeRootDescriptorFileName>
+    <_ILLinkRuntimeRootDescriptorFilePath Condition=" '$(_ILLinkRuntimeRootDescriptorFilePath)' == '' ">$(IntermediateOutputPath)$(_ILLinkRuntimeRootDescriptorFileName)</_ILLinkRuntimeRootDescriptorFilePath>
+    <_NamespaceFilePath Condition=" '$(_NamespaceFilePath)' == '' ">$(MSBuildThisFileDirectory)..\vm\namespace.h</_NamespaceFilePath>
+    <_MscorlibFilePath Condition=" '$(_MscorlibFilePath)' == '' ">$(MSBuildThisFileDirectory)..\vm\mscorlib.h</_MscorlibFilePath>
+    <_CortypeFilePath Condition=" '$(_CortypeFilePath)' == '' ">$(MSBuildThisFileDirectory)..\inc\cortypeinfo.h</_CortypeFilePath>
+    <_RexcepFilePath Condition=" '$(_RexcepFilePath)' == '' ">$(MSBuildThisFileDirectory)..\vm\rexcep.h</_RexcepFilePath>
+  </PropertyGroup>
+
+  <!-- Generates the xml root descriptor file for ILLink. The file contains roots required by the runtime. -->
+  <UsingTask TaskName="CreateRuntimeRootILLinkDescriptorFile" AssemblyFile="$(MSBuildThisFileDirectory)..\..\packages\illink.tasks\$(ILLinkTasksPackageVersion)\tools\net46\ILLink.Tasks.dll" />
```

Comment:
`$(MSBuildThisFileDirectory)..\..\packages\illink.tasks\$(ILLinkTasksPackageVersion)\tools\net46\ILLink.Tasks.dll` that is making a number of bad assumptions.

1. It is assuming the packages directory. Use the PackagesDir property instead.
2. It is assuming we only run on windows on .NET Framework.  We need to be sure this works on .NET Core as well as on non-windows. See https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks.Feed/PackageFiles/Microsoft.DotNet.Build.Tasks.Feed.targets#L42 as an example of how other tasks work.

Reply:
@weshaggard I fixed the issues you pointed out. PTAL.